### PR TITLE
feat: [WD-15550] Adding an sbom workflow

### PR DESCRIPTION
**Pull request by Kxiru**
## Done

- SBOM Generation Workflow. Generates on push.

Fixes [list issues/bugs if needed]

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @mas-who or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](../CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - Push to the branch to trigger the workflow.

## Screenshots

N/A

### CODE REVIEW

File: .github/workflows/sbom_generation.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+    env:
+      TRIVY_VERSION: "0.55.2"
+      TRIVY_ARCH: ${{'64bit'}}
+      TRIVY_SHA256: ${{'09b8730cf16d63b65f1e282f63dfaa16183a143a045b51467864f5e634025a5f' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Comment (mas-who) [inline .github/workflows/sbom_generation.yaml:16]: why is this necessary?

File: .github/workflows/sbom_generation.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        run: |
+          trivy repository . \
+            --format=spdx-json \
+            --output=LXD-UI-SBOM.json
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: sbom-amd64
+          path: "LXD-UI-SBOM.json"
+          retention-days: 7
```
Comment (mas-who) [inline .github/workflows/sbom_generation.yaml:33]: maybe extend this a bit, like 30 days? We will be running this workflow manually and not that frequent.

**Review by mas-who — COMMENTED** [review]
Would you be able to provide a link to a successful action run on your fork for this workflow?

File: .github/workflows/sbom_generation.yaml (hunk 1 above)
Comment (Kxiru) [reply .github/workflows/sbom_generation.yaml:16]: From the [Git checkout action repo](https://github.com/actions/checkout), 

> Only a single commit is fetched by default, for the ref/SHA that triggered the workflow. Set fetch-depth: 0 to fetch all history for all branches and tags. Refer [here](https://docs.github.com/actions/using-workflows/events-that-trigger-workflows) to learn which commit $GITHUB_SHA points to for different events.

I thought that this was important.
